packages/api-server/src/plugins/auth/auth0.ts

```diff
@@ -0,0 +1,109 @@
+import Axios from "axios";
+import { Connection } from "mysql2/promise";
+import { ResultSetHeader } from "mysql2";
+import { DateTime } from "luxon";
+
+import { UserService, UserRole } from "../services/user-service";
+import { APIError } from "../../utils/error";
+
+export interface Auth0UserInfo {
+  sub: string;
+  nickname: string;
+  name: string;
+  picture: string;
+  updated_at: string;
+  email: string;
+  email_verified: boolean;
+}
+
+export async function fetchAuth0UserInfo(
```

Comment:
can we put it in the Auth0UserService? And get the AUTH0_DOMAIN from constructor